api/provider/aws/test_aws/load_balancer_create_test.go

```diff
@@ -0,0 +1,275 @@
+	ingressInput1.SetIpProtocol("TCP")
+	ingressInput1.SetFromPort(int64(80))
+	ingressInput1.SetToPort(int64(80))
+
+	mockAWS.EC2.EXPECT().
+		AuthorizeSecurityGroupIngress(ingressInput1).
+		Return(&ec2.AuthorizeSecurityGroupIngressOutput{}, nil)
+
+	ingressInput2 := &ec2.AuthorizeSecurityGroupIngressInput{}
+	ingressInput2.SetGroupId("lb_sg")
+	ingressInput2.SetCidrIp("0.0.0.0/0")
+	ingressInput2.SetIpProtocol("TCP")
+	ingressInput2.SetFromPort(int64(443))
+	ingressInput2.SetToPort(int64(443))
+
+	mockAWS.EC2.EXPECT().
+		AuthorizeSecurityGroupIngress(ingressInput2).
+		Return(&ec2.AuthorizeSecurityGroupIngressOutput{}, nil)
+
+	iamRoleInput := &iam.CreateRoleInput{}
+	iamRoleInput.SetRoleName("l0-test-lb_id-lb")
+	iamRoleInput.SetAssumeRolePolicyDocument(provider.DefaultAssumeRolePolicy)
+
+	mockAWS.IAM.EXPECT().
+		CreateRole(iamRoleInput).
+		Return(&iam.CreateRoleOutput{}, nil)
+
+	renderedLBRolePolicy, err := provider.RenderLoadBalancerRolePolicy(
+		"region",
+		"123456789012",
+		"l0-test-lb_id",
+		provider.DefaultLBRolePolicyTemplate)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	putIAMPolicyInput := &iam.PutRolePolicyInput{}
+	putIAMPolicyInput.SetPolicyName("l0-test-lb_id-lb")
+	putIAMPolicyInput.SetRoleName("l0-test-lb_id-lb")
+	putIAMPolicyInput.SetPolicyDocument(renderedLBRolePolicy)
+
+	mockAWS.IAM.EXPECT().
+		PutRolePolicy(putIAMPolicyInput).
+		Return(&iam.PutRolePolicyOutput{}, nil)
+
+	serverCertificateMetadata := &iam.ServerCertificateMetadata{}
+	serverCertificateMetadata.SetArn("cert")
+	serverCertificateMetadata.SetServerCertificateName("cert")
+	serverCertificateMetadataList := []*iam.ServerCertificateMetadata{serverCertificateMetadata}
+
+	listServerCertificatesOutput := &iam.ListServerCertificatesOutput{}
+	listServerCertificatesOutput.SetServerCertificateMetadataList(serverCertificateMetadataList)
+
+	mockAWS.IAM.EXPECT().
+		ListServerCertificates(&iam.ListServerCertificatesInput{}).
+		Return(listServerCertificatesOutput, nil).
+		AnyTimes()
+
+	listener1 := &elb.Listener{}
+	listener1.SetProtocol("http")
+	listener1.SetLoadBalancerPort(80)
+	listener1.SetInstancePort(88)
+	listener1.SetSSLCertificateId("cert")
+	listener1.SetInstanceProtocol("http")
+
+	listener2 := &elb.Listener{}
+	listener2.SetProtocol("https")
+	listener2.SetLoadBalancerPort(443)
+	listener2.SetInstancePort(4444)
+	listener2.SetSSLCertificateId("cert")
+	listener2.SetInstanceProtocol("http")
+
+	listeners := []*elb.Listener{listener1, listener2}
+	createLoadBalancerInput := &elb.CreateLoadBalancerInput{}
+	createLoadBalancerInput.SetLoadBalancerName("l0-test-lb_id")
+	createLoadBalancerInput.SetScheme("internet-facing")
+	createLoadBalancerInput.SetSecurityGroups([]*string{aws.String("env_sg"), aws.String("lb_sg")})
+	createLoadBalancerInput.SetSubnets([]*string{aws.String("pub1"), aws.String("pub2")})
+	createLoadBalancerInput.SetListeners(listeners)
+
+	validateFN := func(input *elb.CreateLoadBalancerInput) {
```

Comment:
you can remove this - it's redundant validation with `CreateLoadBalancer(createLoadBalancerInput).` below